Sources/StreamVideo/WebRTC/SFUEventAdapter.swift

```diff
@@ -0,0 +1,379 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+final class SFUEventAdapter {
+
+    var sfuAdapter: SFUAdapter { didSet { observeEvents() } }
+    private let stateAdapter: WebRTCStateAdapter
+    private let disposableBag: DisposableBag = .init()
+    private let recordingUserId = "recording-egress"
+    private let participantsThreshold = 10
+
+    init(
+        sfuAdapter: SFUAdapter,
+        stateAdapter: WebRTCStateAdapter
+    ) {
+        self.sfuAdapter = sfuAdapter
+        self.stateAdapter = stateAdapter
+        observeEvents()
+    }
+
+    private func observeEvents() {
+        disposableBag.removeAll()
+
+        sfuAdapter
```

Comment:
isn't it better to listen to the event wrapper (one of) being published, and do a switch here? Benefits are only one subscription, and making sure we handle all events.